File: frontend/src/types/inventory.ts
export interface InventoryItem {
    inventory_id: string
    id: string
    name: string
    type: string
    rarity: string
    class_type: string
    attack_bonus: number
    defense_bonus: number
    hp_bonus: number
    buy_price: number
    sell_price: number
    level_requirement: number
    equipped_slot: string | null
}

File: backend/app/schemas/inventory.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID

from app.core.enums import ItemType, ItemRarity, ClassType

class InventoryItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    inventory_id: UUID
    id: UUID
    name: str
    type: ItemType
    rarity: ItemRarity
    class_type: ClassType | None
    attack_bonus: int
    defense_bonus: int
    hp_bonus: int
    buy_price: int
    sell_price: int
    level_requirement: int
    equipped_slot: ItemType | None

